### PR TITLE
ClearUnorderedAccessViewUint rework (again)

### PR DESCRIPTION
**Pull request by doitsujin**
Addresses #1069. TL;DR instead of only supporting 32-bit formats for the fallback path, we now use a texel buffer view with the given UINT format to clear the scratch buffer to the desired clear value, then perform a buffer-to-image copy, allowing us to exclude the UINT format from the image's format compatibility list unless otherwise required.

### CODE REVIEW

File: libs/vkd3d/command.c [hunk 1]
```diff
@@ -8307,21 +8307,30 @@ static void d3d12_command_list_clear_uav(struct d3d12_command_list *list,
 
 static void d3d12_command_list_clear_uav_with_copy(struct d3d12_command_list *list,
         const struct d3d12_desc_split *d, struct d3d12_resource *resource,
-        const struct vkd3d_clear_uav_info *args, uint32_t clear_value,
-        UINT rect_count, const D3D12_RECT *rects)
+        const struct vkd3d_clear_uav_info *args, const VkClearColorValue* clear_value,
```
Comment (HansKristian-Work) [inline libs/vkd3d/command.c:8310]: ` *clear_value` nit.

File: libs/vkd3d/command.c [hunk 2]
```diff
@@ -8675,7 +8687,7 @@ static void STDMETHODCALLTYPE d3d12_command_list_ClearUnorderedAccessViewUint(d3
                 /* If the clear color is 0, we can safely use the existing view to perform the
                  * clear since the bit pattern will not change. Otherwise, fill a scratch buffer
                  * with the packed clear value and perform a buffer to image copy. */
-                if (color.uint32[0])
+                if (color.uint32[0] || color.uint32[1] || color.uint32[2] || color.uint32[3])
```
Comment (HansKristian-Work) [inline libs/vkd3d/command.c:8690]: Do we clear unused components to 0 somewhere to avoid hitting fallback due to garbage values?

Comment (HansKristian-Work) [reply libs/vkd3d/command.c:8690]: mask_uint_clear_color should probably clear out unused components?

File: libs/vkd3d/command.c [hunk 3]
```diff
@@ -8550,6 +8543,25 @@ static const struct vkd3d_format *vkd3d_clear_uav_find_uint_format(struct d3d12_
...
+
+    if (resource_format->vk_format == uint_format->vk_format)
+        return true;
+
+    compat = &device->format_compatibility_lists[resource_format->dxgi_format];
+
+    for (i = 0; i < compat->format_count; i++)
+    {
+        if (compat->vk_formats[i] == uint_format->vk_format)
+            return true;
```
Comment (HansKristian-Work) [inline libs/vkd3d/command.c:8559]: I assume TYPELESS will always pass this check?

**Review by HansKristian-Work — CHANGES_REQUESTED** [review]
Small nits, otherwise LGTM.

File: libs/vkd3d/command.c (hunk 1 above)
Comment (doitsujin) [reply libs/vkd3d/command.c]: Fixed.

File: libs/vkd3d/command.c (hunk 3 above)
Comment (doitsujin) [reply libs/vkd3d/command.c:8559]: Yes, all typeless formats have a corresponding UINT format.

File: libs/vkd3d/command.c (hunk 2 above)
Comment (doitsujin) [reply libs/vkd3d/command.c:8716]: Done.

**Review by HansKristian-Work — APPROVED** [review]
LGTM, thanks!

**Pull request merged**